Spec/RealtimeClientPresence.swift

```diff
@@ -1401,6 +1401,164 @@ class RealtimeClientPresence: QuickSpec {
+                            expect(presence.data as? String).to(equal("hello"))
+                            expect(presence.connectionId).to(equal(currentConnectionId))
+                            expect(channelB.presenceMap.members).to(haveCount(2))
+                            expect(channelB.presenceMap.localMembers).to(haveCount(1))
+                            channelB.presence.unsubscribe()
+                            partialDone()
+                        }
+                        channelB.presence.update("hello")
+                    }
+
+                    // LEAVE
+                    waitUntil(timeout: testTimeout) { done in
+                        let partialDone = AblyTests.splitDone(2, done: done)
+                        channelA.presence.subscribe { presence in
+                            guard let currentConnectionId = clientA.connection.id else {
+                                fail("ClientA should be connected"); partialDone(); return
+                            }
+                            expect(presence.action).to(equal(ARTPresenceAction.Leave))
+                            expect(presence.data as? String).to(equal("bye"))
+                            expect(presence.connectionId).toNot(equal(currentConnectionId))
+                            expect(channelA.presenceMap.members).to(haveCount(1))
+                            expect(channelA.presenceMap.localMembers).to(haveCount(1))
+                            channelA.presence.unsubscribe()
+                            partialDone()
+                        }
+                        channelB.presence.subscribe { presence in
+                            guard let currentConnectionId = clientB.connection.id else {
+                                fail("ClientB should be connected"); partialDone(); return
+                            }
+                            expect(presence.action).to(equal(ARTPresenceAction.Leave))
+                            expect(presence.data as? String).to(equal("bye"))
+                            expect(presence.connectionId).to(equal(currentConnectionId))
+                            expect(channelB.presenceMap.members).to(haveCount(1))
+                            expect(channelB.presenceMap.localMembers).to(haveCount(0))
+                            channelB.presence.unsubscribe()
+                            partialDone()
+                        }
+                        channelB.presence.leave("bye")
+                    }
+                }
+
+                // RTP17a
+                it("all members belonging to the current connection are published as a PresenceMessage on the Channel by the server irrespective of whether the client has permission to subscribe or the Channel is configured to publish presence events") {
+                    let options = AblyTests.commonAppSetup()
+                    options.tokenDetails = getTestTokenDetails(capability: "{\"foo\":[\"presence\",\"publish\"]}")
```

Comment:
BTW. I don't think this test will pass now, you may need to add a TODO to fix one @paddybyers releases this functionality to sandbox